addon/utils/djoo.ts

```diff
@@ -0,0 +1,224 @@
+//Dynamic Javascript Orientation Organizer
+// @ts-nocheck
+
+export type PlacementType = 'top' | 'bottom' | 'right' | 'left';
+type ElementOptions = {
+  placement?: string; // the position of the element in relation to the target
+  cssVariableName: string; // name of the custom css variable
+  elementTargetMargin: number; // the margin value between the element and the target
+  viewPortPadding: number; // the padding relative to the viewport
+};
+type ArrowOptions = {
+  defaultRotation: number; // default rotation when placement is top
+  height: number; // total height of the arrow
+  width: number; // total width of the arrow
+};
+
+const CSS_VARIABLE_NAME_PREFIX = '--upf-';
+
+export class Djoo {
+  declare elementOptions: ElementOptions;
+  declare arrowOptions: ArrowOptions;
+
+  hasVerticalAdjustment: boolean = false;
+  hasHorizontalAdjustment: boolean = false;
+
+  computePosition(
+    element: HTMLElement,
+    target: HTMLElement,
+    elementOptions: ElementOptions,
+    arrowOptions: ArrowOptions
+  ): void {
+    this.elementOptions = elementOptions;
+    this.arrowOptions = arrowOptions;
+
+    this.overflowSecurityCheck(element, target);
+
+    const elementTop = this.computedElementTop(element, target);
+    element.style.setProperty(
+      `${CSS_VARIABLE_NAME_PREFIX}${this.elementOptions.cssVariableName}-top`,
+      `${elementTop}px`
+    );
+    const elementLeft = this.computedElementLeft(element, target);
+    element.style.setProperty(
+      `${CSS_VARIABLE_NAME_PREFIX}${this.elementOptions.cssVariableName}-left`,
+      `${elementLeft}px`
+    );
+
+    this.verticalAdjustmentRelativeToViewPort(element);
+    this.horizontalAdjustmentRelativeToViewPort(element);
+
+    if (this.arrowOptions) {
+      const arrowLeft = this.computedArrowLeft(element, target);
+      element.style.setProperty(
+        `${CSS_VARIABLE_NAME_PREFIX}${this.elementOptions.cssVariableName}-arrow-left`,
+        `${arrowLeft}px`
+      );
+      const arrowTop = this.computedArrowTop(element, target);
+      element.style.setProperty(
+        `${CSS_VARIABLE_NAME_PREFIX}${this.elementOptions.cssVariableName}-arrow-top`,
+        `${arrowTop}px`
+      );
+
+      const rotation = this.computedArrowRotation();
+      element.style.setProperty(
+        `${CSS_VARIABLE_NAME_PREFIX}${this.elementOptions.cssVariableName}-arrow-rotation`,
+        `${rotation}deg`
+      );
+    }
+  }
+
+  private overflowSecurityCheck(element, target): void {
```

Comment:
"check" in the name is not exact, this method actually assigns a new position if needed.

This is of course nit-picking ^^

how about returning the new position and assigning it in the parent ?
This way the current method name makes sense

Reply:
This method only updates the placement. I change the name for `overflowPlacementCorrection`